### ISSUE TITLE
[Feature Request] Support for converting both ways

### ISSUE THREAD

**Issue by EriKWDev** [alex-myznikov/from_css_color#1]
The package works great, but it would be nice to be able to convert from a Dart Color into the various css formats :)

I imagine something like `String toCssHex(<Color color>)` would return a string on the format "#ffffffff", `String toCSSRGB(<Color color>)` would turn it into "rgb(255, 255, 255)" and so on..

**Comment by alex-myznikov** [comment]
Hello!

I'm starting to think of some kind of extension for the `Color` type that should be reasonably safe in terms of naming conflicts and allow an unlimited number of formats to be added, for example: hex, hexShort, hexAlpha, hexAlphaShort (https://drafts.csswg.org/css-color-3 already defines quite enough of various formats allowed and this amount may grow up in upcoming proposals). I'll schedule this work for the next weekend.

Until then, could you provide some more details on your use cases to help projecting for the requested feature?

**Comment by EriKWDev** [comment]
Yeah, an extension on the Color would be amazing! Color.fromCss() and then on a color instance color.toLongHex().

My use case is that I have a textfield where the user can input a color and the input format accepts all CSS input formats. 

I am then programmatically generating a color palette from the input and would like to display it to the user again so that they can copy the generated colors and use in a project. (A color palette generator basically).

It would be nice if I could let the user choose an output format of these colors and get them as a hex string, rgba string, hsv string and so on.

**Comment by alex-myznikov** [comment]
Ok, thanks for details:+1:

I've planned work on the update and will probably come up with it by the end of the next week or so...

**Comment by EriKWDev** [comment]
Here is how I currently convert from Dart's own `Color` into a hex string with rgba:
```dart
'#${color.red.toRadixString(16).padLeft(2, '0')}${color.green.toRadixString(16).padLeft(2, '0')}${color.blue.toRadixString(16).padLeft(2, '0')}${color.alpha.toRadixString(16).padLeft(2, '0')}';
```
where `padLeft` inserts leading zeros and `toRadixString` converts a decimal number to a string representation of that number in a given base.

**Comment by alex-myznikov** [comment]
I have released version 1.1.0 with
```dart
String toCssString({CssColorString format = CssColorString.hex})
```
extension method on `Color`, returning CSS color string representation.

To implement a static extension method that creates `Color` instance from CSS string I need this feature https://github.com/dart-lang/language/issues/723 to be implemented.

I also have not added anything but basic `CssColorString.hex` and `CssColorString.rgb` formats for conversion, but they are designed to be smart enough and return the shortest possible yet detailed string representation of a `Color` instance (i.e. `CssColorString.hex` can return 3/4/6/8-digit hex representations where appropriate as well as `CssColorString.rgb` can return both rgb/rgba). Examples and tests cover the named use-cases pretty decently.

If you have more concrete use-cases where some specific formats are desirable - feel free to open a new issue for me to add them.
